modules/cas_cache/layer_cache_management.c

```diff
@@ -406,20 +422,24 @@ struct _cache_mngt_attach_context {
 };
 
 
-static void cache_start_rollback(struct work_struct *work)
+static int cache_start_rollback(void *data)
 {
 	struct cache_priv *cache_priv;
 	ocf_queue_t mngt_queue = NULL;
-	struct _cache_mngt_attach_context *ctx =
-		container_of(work, struct _cache_mngt_attach_context, work);
+	struct _cache_mngt_attach_context *ctx = data;
 	ocf_cache_t cache = ctx->cache;
 	int result;
 
+	if (kthread_should_stop())
+		return 0;
+
 	if (ctx->cls_inited)
 		cas_cls_deinit(cache);
 
 	if (ctx->priv_inited) {
 		cache_priv = ocf_cache_get_priv(cache);
+		kthread_stop(cache_priv->stop_context->finish_thread);
+		kfree(cache_priv->stop_context);
```

Comment:
cache_priv->stop_context is initialized inside _cache_mngt_cache_priv_init(), so it would make sense to deinitialize it inside _cache_mngt_cache_priv_deinit().